timeline 컴포넌트 구조 잡기

beomseok37 opened the pull request:
### 제목
timeline 구조 잡기

### 파일 변경
-  frontend/src/components/Header/index.tsx : 컴포넌트명 수정
- frontend/src/components/Post/Content/index.tsx : 컴포넌트 추가
-  frontend/src/components/Post/Content/style.ts : 컴포넌트 추가
-  frontend/src/components/Post/Image/index.tsx : 컴포넌트 추가
-  frontend/src/components/Post/Image/style.ts : 컴포넌트 추가
-  frontend/src/components/Post/LikeRow/index.tsx : 컴포넌트 추가
-  frontend/src/components/Post/LikeRow/style.ts : 컴포넌트 추가
-  frontend/src/components/Post/Review/index.tsx : 컴포넌트 추가
-  frontend/src/components/Post/Review/style.ts : 컴포넌트 추가
-  frontend/src/components/Post/index.tsx : post 컴포넌트 구조 잡기
-  frontend/src/components/Post/style.ts : post 컴포넌트 구조 잡기
-  frontend/src/components/Timeline/index.tsx : timeline 컴포넌트 구조 잡기
-  frontend/src/components/Timeline/style.ts : timeline 컴포넌트 구조 잡기
-  frontend/src/components/buttons/{LogoButton → ImageButton}/index.tsx  : 컴포넌트 명 수정
- frontend/src/components/buttons/{LogoButton → ImageButton}/style.ts : 컴포넌트 명 수정
-  frontend/src/components/buttons/ProfileImageButton/index.tsx : 컴포넌트 추가
-  frontend/src/components/buttons/ProfileImageButton/style.ts : 컴포넌트 추가
-  frontend/src/components/buttons/post/CommentButton/index.tsx : 컴포넌트 추가
-  frontend/src/components/buttons/post/CommentButton/style.ts : 컴포넌트 추가
-  frontend/src/components/buttons/post/EchoButton/index.tsx : 컴포넌트 추가
-  frontend/src/components/buttons/post/EchoButton/style.ts : 컴포넌트 추가
-  frontend/src/components/buttons/post/LikeButton/index.tsx : 컴포넌트 추가
-  frontend/src/components/buttons/post/LikeButton/style.ts : 컴포넌트 추가

beomseok37 committed:
Refactor: LogoButton 이름 ImageButton으로 수정

beomseok37 committed:
Feat: timeline post에 사용될 button component 구조 잡기

beomseok37 committed:
Feat: timeline post에 사용될 component 구조 잡기

beomseok37 committed:
Feat: timeline, post component 구조 잡기

beomseok37 committed:
Style: Post Wrapper sytle적용 button background style 적용

- post wrapper padding:50px 0, border-radius:50px 적용
- button background: #444444 적용

beomseok37 committed:
Refactor: post 관련 component 이름 수정

- 폴더명에 Post 삭제

hyunmindev commented on frontend/src/components/Post/index.tsx:
```diff
@@ -0,0 +1,38 @@
+    <Wrapper>
+      <ProfileImageButton href='users/123'>image, username</ProfileImageButton>
+      <PostImage />
+      <IconRow>
```
기존에 정의해두었던 Grid 스타일 컴포넌트를 활용하는건 어떨까요!
?

beomseok37 replied:
row관련  style말고도 margin-left와 svg간 margin도 있어서 새로 만들었어요!!

hyunmindev replied:
`Row` 의 활용도가 낮아지는게 우려됩니다
`Row` 컴포넌트 내부에서 `Buttons`와 같은 새로운 컴포넌트를 정의하는건 어떨까요

hyunmindev commented on frontend/src/components/Timeline/index.tsx:
```diff
@@ -1,5 +1,16 @@
-  return <>Timeline</>;
+  return (
+    <Wrapper>
+      <Card width={600} height={500}>
```
width값은 더 크게 주어야할 것 같습니다 
정확한 값은 추후에 다시 상의해봐요~ 일단은 pass!

beomseok37 replied:
넵!!

hyunmindev reviewed (commented):
코멘트 하나 확인해주세요~

dmin0211 commented on frontend/src/components/Post/Content/style.ts:
```diff
@@ -0,0 +1,8 @@
+import styled from '@emotion/styled';
+
+const Wrapper = styled.div`
+  margin-left: 15px;
+`;
+
+// eslint-disable-next-line import/prefer-default-export
```
style.ts 가 다중의 style 컴포넌트를 반환할 것을 예상하여 export 로 사용하신다고 했는데 이름은 wrapper 로 통일하시는 건가요?

hyunmindev replied:
네 껍데기 컴포넌트는 Wrapper로 통일했습니다

beomseok37 replied:
Wrapper로 통일한 것은 아니고 저 컴포넌트 전체를 감싼다는 의미로 사용하였습니다.

dmin0211 replied:
컴포넌트의 껍데이인 Wrapper 이외에 다른 style은 어떤게 있을까요?? 또한 그 style 의 naming 은 어떻게 되는 건가요?? 순수한 궁금점입니다!

hyunmindev replied:
src/components/Header/style.ts 파일에 스타일 컴포넌트를 두개 정의한 것을 볼 수 있습니다
지금은 스타일을 크게 신경쓰지않고 프로토타입을 잡아가는 위주로 개발중이다보니 대부분 Wrapper 스타일 컴포넌트만 필요하지만 추후에 스타일을 상세하게 잡을 시 여러개의 Styled component가 필요할 것으로 예상돼요

youjeonghan reviewed (approved):
수고하셧습니다~

dmin0211 commented on frontend/src/components/buttons/post/LikeButton/style.ts:
```diff
@@ -0,0 +1,9 @@
+import styled from '@emotion/styled';
+
+const Button = styled.button`
+  background: #444444;
```
이런 색들은 한곳에 모아서 관리하는게 어떨까요??

hyunmindev replied:
네 styles/theme.ts에 테마 정의하는 곳 있습니다 emotion theming 으로 검색 후 사용해보세요

beomseok37 replied:
background: transparent로 사용하면 되어서 transparent로 고쳤습니당

dmin0211 replied:
네 알겠습니다~~ 그래도 theme 에는 색을 넣어주세요!!

hyunmindev replied:
@beomseok37 theme에 #444444와 같은 자주 사용하는 색 정의 부탁드릴게요

beomseok37 committed:
Style: button background 투명하게 수정

dmin0211 reviewed (approved):
승인합니다,

beomseok37 committed:
Chore: gitignore에 .env 추가

beomseok37 committed:
Refactor: button background, border 속성 global css로 적용

beomseok37 committed:
Refactor: 리뷰 반영

- Row component style 적용
- 전체 스타일링을 위해 icon button을 div로 묶음

beomseok37 committed:
Refactor: 리뷰 반영

- Row component에 justify content props 추가
- Row component사용하는 component에 props 추가

hyunmindev commented on frontend/src/components/Grid/index.ts:
```diff
@@ -1,10 +1,14 @@
   flex-direction: row;
   align-items: center;
-  justify-content: center;
+  justify-content: ${({ justifyContent }) => justifyContent};
```
justifyContent 값이 props로 들어오지 않을 시 default Value를 넣어놓는 것이 편하겠어요!

beomseok37 committed:
Refactor: Row component default props 정의

- justify-content의 속성이 안 들어올 경우 center로 적용

hyunmindev reviewed (approved):
고생하셨습니다!
